src/Aminal.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import "openzeppelin/token/ERC721/ERC721.sol";
+
+error AminalDoesNotExist();
+error SenderDoesNotHaveMaxAffinity();
+error ExceedsMaxLocation();
+error OnlyMoveWithGoTo();
+error MaxAminalsSpawned();
+
+contract Aminal is ERC721 {
+    constructor() ERC721("Aminal", "AMNL") {}
+
+    uint160 constant MAX_LOCATION = 1e9;
+
+    uint256 constant MAX_AMINALS = 1e4;
+
+    uint256 currentAminalId;
+
+    bool going;
+
+    modifier goingTo() {
+        going = true;
+        _;
+        going = false;
+    }
+
+    event AminalSpawned(
+        address spawner,
+        uint256 aminalId,
+        uint256 value,
+        uint256 affinity
+    );
+
+    event AminalFed(
+        address feeder,
+        uint256 aminalId,
+        uint256 value,
+        uint256 newAffinity,
+        bool newMax
+    );
+
+    mapping(uint256 => mapping(address => uint256)) public affinity;
+    mapping(uint256 => uint256) public maxAffinity;
```

Comment:
I read this as the max affinity that an aminal could reach. Is this the highest affinity?